project/frontend/src/components/ProfileForm.js

```diff
@@ -1,45 +1,107 @@
-import React from 'react';
 import './ProfileForm.css';
 
-import PreferenceForm from './PreferenceForm.js';
+import React, { useContext, useState } from 'react';
+
+import { AuthContext } from '../components/Authentication.js';
 import ButtonGroup from './ButtonGroup';
+import FirestoreContext from '../contexts/FirestoreContext.js';
+import PreferenceForm from './PreferenceForm.js';
 
 /**
  * A form used to gather information for a user's profile page.
  */
 function ProfileForm() {
+  const { firestore } = useContext(FirestoreContext);
+  const authContext = useContext(AuthContext);
+  let user = authContext.currentUser.get;
+
   const itemLabels = {
     cuisine: 'Preferred Cuisine',
     location: 'Your Location',
     distance: 'Preferred Distance',
     price: 'Price Level',
     experience: 'Dining Experience',
   };
-  const cuisineOptions = ['Italian', 'Mexican'];
-  // This is a filler value. I plan on moving all cuisine stuff to PreferenceForm.js
-  const setCuisine = () => {};
+  const labelsList = {
+    distance: ['1 mile', '5 miles', '10 miles', '25 miles'],
+    price: ['$', '$$', '$$$', '$$$$'],
+    experience: ['Takeout', 'Delivery', 'Dine-In'],
+  };
+  const cuisineOptions = [''];
+
+  const [selectedCuisines, setSelectedCuisines] = useState([]);
+  const [clickedDistanceButtons, setClickedDistanceButtons] = useState([]);
+  const [clickedPriceButtons, setClickedPriceButtons] = useState([]);
+  const [clickedExperienceButtons, setClickedExperienceButtons] = useState([]);
+
+  /**
+   * A callback function that is going to be used to get the updated
```

Comment:
nit: 'that is used to get the updated...' instead of 'going to be used'
and also line 41: Updates the clicked state for the given id. 

(last writing clear comments nit haha)